libs/src/utils/network.js

```diff
@@ -58,7 +59,7 @@ async function raceRequests (
     // 1. We try public gateways first
     // 2. We give requests the opportunity to get canceled if other's are very fast
     await Utils.wait(timeBetweenRequests * i)
-
+    if (hasFinished) return
```

Comment:
Does having undefined in the requests array make any issues in promiseFight?

Reply:
nice catch. I think we should do a good ole filter(Boolean)

Reply:
I don't think we need to do anything here b/c this is in a promise race and if this is true that mean a promise resolved.

Reply:
Yeah I feel good about this too :) https://repl.it/repls/FreeScratchyPaint#index.js